codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -1394,6 +1395,101 @@ getDependencyIdsForDependent dependent@(C.Reference.Id oid0 _) =
     isNotSelfReference (C.Reference.Id oid1 _) =
       oid0 /= oid1
 
+-- | Given two term (components) A and B, return the set of all terms that are along any "dependency path" from A to B,
+-- not including A nor B; i.e., the transitive dependencies of A that are transitive dependents of B.
+--
+-- For example, if A depends on X and Y, and Y depends on Z, and X and Z depend on B...
+--
+--     --X--
+--    /     \
+--   A       B
+--    \     /
+--     Y---Z
+--
+-- ...then `getDependenciesBetweenTerms A B` would return the set {X Y Z}
+getDependenciesBetweenTerms :: ObjectId -> ObjectId -> Transaction (Set ObjectId)
+getDependenciesBetweenTerms oid1 oid2 =
+  queryListCol sql (oid1, oid2) <&> Set.fromList
+  where
+    -- Given the example above, we'd have tables that look like this.
+    --
+    -- First, the `paths` table finds all paths from source `A` to sink `B`, exploring depth-first. As a minor
+    -- optimization, we seed the search not with `A`, but rather the direct dependencies of `A` (namely `X` and `Y`).
+    --
+    -- +-paths-------------------------------+
+    -- +-level-+-object_id-+-object_id_array-+
```

Comment:
Maybe easier to understand if `object_id_array` was called `path`? or `path_so_far` Or maybe we can talk through it on Weds :)

Reply:
I think I made this better, see [58e2bf6](https://github.com/unisonweb/unison/pull/3466/commits/58e2bf6541f2d3b226ec11eca1488bb227f71a19)